components/message_queue/src/kafka/kafka_impl.rs

```diff
@@ -0,0 +1,396 @@
+use std::{cmp::Ordering, fmt::Display, sync::Arc};
+
+use async_trait::async_trait;
+use common_util::define_result;
+use dashmap::DashMap;
+use futures::StreamExt;
+use log::info;
+use rskafka::{
+    client::{
+        consumer::{StartOffset, StreamConsumer, StreamConsumerBuilder},
+        controller::ControllerClient,
+        error::{Error as RskafkaError, ProtocolError},
+        partition::{Compression, OffsetAt, PartitionClient},
+        Client, ClientBuilder,
+    },
+    record::{Record, RecordAndOffset},
+};
+use snafu::{ensure, ResultExt, Snafu};
+
+use super::config::WalConfig;
+use crate::{kafka::config::Config, ConsumeAllIterator, Message, MessageAndOffset, MessageQueue};
+
+/// The topic(with just one partition) client for Kafka
+//
+/// `Arc` is needed to ensure its lifetime. Because in future's gc process,
+/// it may has removed from pool but still is still being used.
```

Comment:
```suggestion
/// The topic (with just one partition) client for Kafka
//
/// `Arc` is needed to ensure its lifetime because in future's gc process,
/// it may has removed from pool but be still in use.
```